### PR TITLE
update volunteer engineer signup form

### PR DESCRIPTION
**Pull request by tanyaparker**
[original spec](https://docs.google.com/document/d/1tNjrUcbdUbhUTenIAvu7HpDmrw5G-OmKDrGLo7d3VzU/edit#heading=h.s40u2j1i7kj6)

image

### CODE REVIEW

**Comment by tanyaparker** [comment]
@laurelfan I have a database question. I want to make the last checkbox required. There may be existing people who haven't checked this before and are nil. So I need to transfer them to the Engineer mailchimp mailing list first, and then make the checkbox required. How can I make sure that nobody signs up between this change? Is there a way I can block signups?

**Comment by tanyaparker** [comment]
@laurelfan Also related, I'm changing some of the fields and will need to update past sign ups. Is there a wiki for pegasus migrations?

File: pegasus/sites.v3/code.org/views/volunteer_engineer_form.haml [hunk 1]
```diff
@@ -19,117 +20,118 @@
...
-    %button#btn-submit.btn.btn-default{type: "submit"} Submit
+    %button#btn-submit.btn.btn-default{type: "submit"}= I18n.t(:volunteer_engineer_submission_submit)
 
 #thanks{:style=>"display: none;"}
-  %p Thank you for submitting your information.
+  %p= I18n.t(:volunteer_engineer_submission_thankyou)
 
 :javascript
   $(document).ready(function() {
+    new google.maps.places.SearchBox(document.getElementById('volunteer-location'));
```
Comment (CaptainStack) [inline pegasus/sites.v3/code.org/views/volunteer_engineer_form.haml:137]: Would it maybe be better to use jQuery here for cross browser compatibility?

**Comment by CaptainStack** [comment]
Left one line note. Didn't have a lot to say besides that. There's a lot of stuff here, so I'm sure it goes without saying that you've tested it a lot. Did you try all the combinations of incomplete data? Did you check how the data looks in the db? If you've done all that, then I'm sure it's good to go.

**Comment by tanyaparker** [comment]
Added images to top 

image

**Comment by CaptainStack** [comment]
> @laurelfan Also related, I'm changing some of the fields and will need to update past sign ups. Is there a wiki for pegasus migrations?

I don't think there is (but there should be). I was told to just make it manually following the naming convention of the ones already there. You can run it locally with rake db:migrate, and like on Dashboard, it's supposed to run automatically on build once it's in staging.

**Comment by laurelfan** [comment]
Hm, would it make more sense to add a version field to the new form (or even make a new form type) if we want to treat the new data differently than the old data?

Eg. The 2015 hoc sign up is a different form than the 2014 hoc sign up... We can query from both when we want to use the sign ups together, and we can query from just one when we only want that.

Adding a version will also allow you to do the migration later without having to prevent people from using the new form.

**Pull request merged**